data_model/src/transaction.rs

```diff
@@ -149,38 +150,26 @@ impl Txn for VersionedTransaction {
     #[inline]
     fn payload(&self) -> &Payload {
         match self {
-            Self::V1(v1) => &v1.0.payload,
-        }
-    }
-}
-
-impl PartialEq for VersionedTransaction {
-    fn eq(&self, other: &Self) -> bool {
-        use VersionedTransaction::*;
-
-        match (self, other) {
-            (V1(first), V1(second)) => first.0.eq(&second.0),
+            Self::V1(v1) => &v1.payload,
         }
     }
 }
 
 impl From<VersionedValidTransaction> for VersionedTransaction {
     fn from(transaction: VersionedValidTransaction) -> Self {
         match transaction {
-            VersionedValidTransaction::V1(v1) => {
-                let tx: ValidTransaction = v1.0;
-
-                let signatures = tx
+            VersionedValidTransaction::V1(transaction) => {
+                let signatures = transaction
                     .signatures
                     .values()
                     .iter()
                     .cloned()
                     .collect::<BTreeSet<_>>();
-                let tx = Transaction {
-                    payload: tx.payload,
+                let transaction = Transaction {
```

Comment:
```suggestion
                let transaction = Transaction {
```
This shadowing is unnecessary.

Reply:
I don't see a difference here